nemo/collections/asr/models/aed_multitask_models.py

```diff
@@ -224,6 +224,9 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
             self.decoding, tokenize=self.cfg.get('bleu_tokenizer', "13a"), log_prediction=False
         )  # Wer is handling logging
 
+        # Setup encoder adapters (from ASRAdapterModelMixin)
```

Comment:
couldn;t find how its calling set up of encoder adapter, could you please shed some light here on how this is calling specifically "encoder" layer adapter.

Reply:
This is the core entrypoint of adapters. 
Users should mostly focus on https://docs.nvidia.com/nemo-framework/user-guide/latest/nemotoolkit/core/adapters/intro.html#adapters and https://docs.nvidia.com/nemo-framework/user-guide/latest/nemotoolkit/core/adapters/intro.html#using-the-adapter-module

Devs should mostly focus on https://colab.research.google.com/github/NVIDIA/NeMo/blob/stable/tutorials/02_NeMo_Adapters.ipynb

I think it would be good to link the dev tutorial somewhere in the core docs.

Reply:
As to your question about "specific encoder adapter" - that's not this classes responsibility but the `ASRModuleMixin` which inherits `ASRAdapterModelMixin`. Hmm we need some tool to generate class diagrams.. 

In any case, ASRAdapterModelMixin determines how to direct the `adapter_name` (`encoder:xyz`) to the module. See below definition of `ASRAdapterModelMixin`.